src/main/sd/init.c

```diff
@@ -0,0 +1,310 @@
+#define COMMAND_READ_CSD            (uint8_t)0x09    //CMD09
+
+#define SD_INIT_OK              (uint8_t)0x00
+#define SD_INIT_NOK             (uint8_t)0x01
+#define SD_IDLE                 (uint8_t)0x02
+#define SD_ERROR                (uint8_t)0x03
+#define SD_INITIALIZED          (uint8_t)0x04
+
+#define SD_CONFIGURE_PINS       (uint8_t)0x05
+#define SD_RESET_SOFTWARE       (uint8_t)0x06
+#define SD_CHECK_VOLTAGE        (uint8_t)0x07
+#define SD_READ_OCR             (uint8_t)0x08
+#define SD_PREPARE_INIT_PROCESS (uint8_t)0x09
+#define SD_START_INIT_PROCESS   (uint8_t)0x0a
+#define SD_READ_OCR_AGAIN       (uint8_t)0x0b
+#define SD_READ_CSD             (uint8_t)0x0c
+
+bool check_sd_version = false;
+uint8_t csd_register[16];
+
+bool reset_software()
+{
+    //set CMD0
+    uint8_t command_argument[] = {0x00, 0x00, 0x00, 0x00};
+    set_command(command, COMMAND_SOFTWARE_RESET, command_argument); 
+
+    mext2_response* response = send_command(command, MEXT2_R1);
+    if(response == NULL || response -> r1 != R1_IN_IDLE_STATE)
+    {
+        return false;
+    }
+
+    wait_8_clock_cycles((uint8_t*) response);
+    return true;
+}
+
+bool check_voltage_range()
+{
+    //set CMD8
+    uint8_t command_argument[] = {0x00, 0x00, 0x01, 0xaa};
+    set_command(command, COMMAND_CHECK_VOLTAGE, command_argument); 
+
+    mext2_response* response = send_command(command, MEXT2_R7);
+     if(response == NULL)
+    {
+        return false;
+    }
+
+    if(response -> r1 & R1_ILLEGAL_COMMAND)
+        sd_version = SD_V1X;
+
+    wait_8_clock_cycles((uint8_t*) response);
+    return true;
+}
+
+bool read_OCR()
+{
+    //set CMD58
+    uint8_t command_argument[] = {0x00, 0x00, 0x00, 0x00};
+    set_command(command, COMMAND_READ_OCR, command_argument); 
+
+    mext2_response* response = send_command(command, MEXT2_R3);
+    if(response == NULL || (response -> r1 & R1_ILLEGAL_COMMAND) == true)
+    {
+        return false;
+    }
+
+    if(check_sd_version)
+    {
+        if(response -> ocr[0] & (uint8_t)0x40)
+            sd_version = SD_V2XHCXC;
+        else
+            sd_version = SD_V2X;
+    }
+
+    wait_8_clock_cycles((uint8_t*) response);
+    return true;
+}
+
+bool prepare_init_process()
+{
+    //set CMD55
+    uint8_t command_argument[] = {0x00, 0x00, 0x00, 0x00};
+    set_command(command, COMMAND_BEFORE_INIT_PROCESS, command_argument); 
+
+    mext2_response* response = send_command(command, MEXT2_R1);
+    if(response == NULL || response -> r1 & R1_ILLEGAL_COMMAND)
+    {
+        return false;
+    }
+    wait_8_clock_cycles((uint8_t*) response);
+    return true;
+}
+
+uint8_t start_init_process()
+{
+    //set ACMD41
+    uint8_t command_argument[] = {0x00, 0x00, 0x00, 0x00};
+    set_command(command, COMMAND_INIT_PROCESS, command_argument); 
+
+    mext2_response* response = send_command(command, MEXT2_R1);
+    if(response == NULL)
+    {
+        return 0;
+    }
+    
+    return response -> r1;
+}
+
+
```

Comment:
Two newlines between functions when everywhere else we have one

Reply:
corrected